src/avram/type.cr

```diff
@@ -21,6 +21,10 @@ module Avram::Type
     end
   end
 
+  def parse(value)
+    FailedCast.new
+  end
```

Comment:
This was added because the update to `Avram::Criteria` uses a method that can return a bunch of types. I think a downside of this is that the `parse` methods are how you limit the types that can be passed to query methods which means that any time can be passed into query methods now and they will raise a runtime error

It might be better to add an overload of `parse!` to handle this so that we can keep the type safety at the lower level. It kind of makes sense that the bang method is less safe as well